app/frontend/terrier/parts/select-field-part.ts

```diff
@@ -0,0 +1,214 @@
+     * @param dropdown the element to reposition
+     * @param selectField the anchor select field
+     * @param selectedElement the selected option that will be positioned directly above the select field
+     */
+    anchorSelectDropdown(dropdown: HTMLElement, selectField: HTMLElement, selectedElement: HTMLElement) {
+        const dropdownSize = Overlays.getElementSize(dropdown)
+        const selectOptionSize = Overlays.getElementSize(selectedElement)
+
+        const selectedOptionIndex = Array.prototype.indexOf.call(dropdown.childNodes, selectedElement)
+
+        const anchorRect = selectField.getBoundingClientRect()
+        const win = Overlays.getWindowSize()
+
+        const result: AnchorResult = {
+            top: anchorRect.y - (selectOptionSize.height * selectedOptionIndex),
+            left: anchorRect.x,
+            valid: true
+        }
+
+        Overlays.clampAnchorResult(result, dropdownSize, win)
+
+        if (dropdownSize.width < anchorRect.width) {
+            result.width = anchorRect.width
+        }
+
+        // how far down the select options list we need to scroll
+        let scrollAmount = (selectOptionSize.height * (selectedOptionIndex))
+        const hasOptGroups = this.state.options.find((o) => 'group' in o)
+        if (hasOptGroups) {
+            scrollAmount -= selectOptionSize.height
+        }
+
+        let scrollTop = scrollAmount - anchorRect.top
+        let dropdownIsShrunk = false
+
+        if (dropdownSize.height > win.height - anchorRect.top) { // attempting to fill the whole window height with the dropdown, so we may need to adjust placement of dropdown
+            if (scrollAmount < anchorRect.top) { // need space above the dropdown
+                result.top = anchorRect.top - scrollAmount
+                result.height = win.height - result.top
+                scrollTop = 0
+                dropdownIsShrunk = true
+            } else if (dropdownSize.height - scrollAmount < win.height) { // need space below the dropdown
+                result.height = dropdownSize.height - scrollAmount + anchorRect.top
+                scrollTop = scrollAmount
+                dropdownIsShrunk = true
+            }
+        } else {
+            if (hasOptGroups) {
+                result.top += selectOptionSize.height
+            }
+        }
+
+        let styleString = ""
+        for (const key of ['top', 'left', 'width', 'height'] as const) {
+            if (result[key] != null) {
+                styleString += `${key}: ${result[key]}px;`
+            }
+        }
+
+        dropdown.setAttribute('style', styleString)
+        dropdown.scrollTo({ top: scrollTop })
+
+        dropdown.classList.add('show')
+        selectedElement.classList.add('hover')
+        selectedElement.addEventListener('mouseleave', () => selectedElement.classList.remove('hover'))
+
+        // if the dropdown has been temporarily shrunk, increase the size as we scroll
+        if (dropdownIsShrunk) {
```

Comment:
I'm not sure putting all of this effort into making scrolling work similarly to the native mac dropdowns is really worth it. Especially if we're doing it for familiarity's sake, most of our customers use windows so we're not really gaining anything in terms of familiarity with them. I think it looks nice and matches the mac behavior pretty well but I don't think the amount of code required to implement it is worth it, especially for a purely visual effect.